Sabbat, Vaulderie, Antag Rework, Sabbat Lair, Ritae

chazzyjazzy opened the pull request:
## About The Pull Request

This PR aims to improve and expand upon the Sabbat antagonist experience in TFN by introducing them as their own roundstart sect. It introduces three roundstart roles, Sabbat Pack (2 slots), Sabbat Priest (1 slot), Sabbat Ductus (1 slot), as well as their own haven where they dwell, in the abandoned big building just south of the park. It also introduces multiple ritae and rituals the Sabbat often use - introducing functionality for Vaulderie and blood bonding via the Silver Goblet in the Prince's office or the Vaulderie Goblets found in the Sabbat Lair, the rite of Monomacy which allows Sabbat packmates to challenge each other to duels if they cannot settle disputes peacefully, a war party totem which summons all sabbat in the city (with decreasing reach for every use of the totem) to the lair in order to go on a hunt of some sort, a campfire inside the Sabbat lair so that Sabbat members may run through the campfire (which is apart of their initiation ritae), the ability to smack Kindred over the head with a shovel and bury them in a grave, where, 120 seconds later, the will dig themselves out of the grave and be in a 30 second straight frenzy, as well as the 'blood bath' supervized by the Sabbat Priest which allows for a new Ductus to be chosen. 

What still needs to be done:
- restrict roles to whitelist-only
- introduce a 'sabbat priest's tome' which lists out the ritae and how to perform them
- more testing would be nice (at least it compiles)
- blood bath ritae needs to transfer Ductus antag datum but only if the priest allows it 

## Why It's Good For The Game

for a long time now the antag experience on this codebase has been almost literally nonexistent, people dont play as antag since its ss13 vibes running up to people and saying 'sword of caine' click click horizontal which isnt fun or interesting for anyone. We need more interesting conflict and more 'villainous' anti-vampire factions to make the experience more dangerous and threatening. Hounds should actually have to protect the Elders, not just attack breachers. Primogen shouldnt feel safe just wandering around in public or hanging out in a bar. The Sabbat and the conflict with them is a core part of the setting and its time we give players the ability to play round-persistent sabbat characters, with the caveat that these roles are not just fragmachines (as there is a disclaimer when you spawn in with the role) and flavor to occupy Sabbat members on their own time. 

## Testing Photographs and Procedure

https://www.youtube.com/watch?v=7BGaMaj3GzI


## Changelog

:cl:
add: Adds Sabbat sect with three playable roles, Sabbat Pack, Sabbat Ductus, Sabbat Priest.
add: Multiple ritae and flavor for Sabbat, including Vaulderie, war parties, initiation rites, shovelheading, monomacy, and blood baths
add: Players can now select an 'Antagonist Name' on their character sheet so that they don't have to spawn in with a random name when playing lateparty antags like hunters or Sabbat
bugfix: fixed heart sprites disappearing on the floor (sorry for making this part of this PR - its a one line change)
del: Removed redundant "Caitiff" lateparty setting, as we have Caitiff available for round-persistent characters already
map: Adds a run-down Sabbat lair, home to only the most beastly of Cainites.

/:cl:

chazzyjazzy commented:
another caveat - i'm generally aware that introducing a permanent antag faction to both the camarilla and the anarchs will boil most rounds down to 'whos the sabbat - theres the sabbat get them!!!!' instead of the more nuanced and interesting clan rp that we have had in the past, which is why i think it would be ideal to introduce hunters roundstart, or perhaps limit the Sabbat even further in some way, though they already have very few slots.

CalamaBanana commented:
> another caveat - i'm generally aware that introducing a permanent antag faction to both the camarilla and the anarchs will boil most rounds down to 'whos the sabbat - theres the sabbat get them!!!!' instead of the more nuanced and interesting clan rp that we have had in the past, which is why i think it would be ideal to introduce hunters roundstart, or perhaps limit the Sabbat even further in some way, though they already have very few slots.

As a Camarilla RP player, that one is my worry as well. My suggestion on discord was to cripple the Sabbat with rules just a bit. It needs to be enforced that they have to show a minimum value for their own life, so we do not get the funny man with a flamethrower suiciding into each and every Camarilla meeting every round.

chazzyjazzy commented:
changes since the creation of the PR:

added a sabbat priest's tome which shows a tgui input list where the priest ONLY may select ritae which prints descriptions of what they are and how to do them. only the priest may see these. The priest may also edit a default interpretation of the Sabbat's credo (as the sabbat have no official written records according to wiki), which allows for different player priests to change the mission statement of the pack in the city. perhaps some priests are more warlike, some less so. perhaps some priests are more theological and religious, perhaps some are more secular and intellectual. changing the credo allows for some flexibility there, but I think the default credo is a good catch-all. 

i've also made the roles whitelist only with 'whitelisted = TRUE'

i've also finished the blood bath rite

this pr is now ready for code review

Kevak commented:
I can't say I'm particularly pleased with how fast burying for the shovelhead ritual is. Have you ever buried a body? Takes a good bit! Ten seconds minimum, not barely a second or two. Adjust such. Additionally, consider moving their haven further south into the slum zone. Below the park is right next door to a garou den, the lasombra haven, and the ventrue haven. No way the sabbat could set up there unnoticed

Magisterium2022 commented:
Concept-wise this seems all good to me - just make sure the role spawn text makes it clear that they're not there to murderbone or ruin rounds.

Kevak commented:
> > I can't say I'm particularly pleased with how fast burying for the shovelhead ritual is. Have you ever buried a body? Takes a good bit! Ten seconds minimum, not barely a second or two. Adjust such. Additionally, consider moving their haven further south into the slum zone. Below the park is right next door to a garou den, the lasombra haven, and the ventrue haven. No way the sabbat could set up there unnoticed
> 
> It was already at ten seconds. Despite that, I've upped it to 20. Also, there is no slum area further south of that point. Theres a tunnel with a homeless encampment and then the forest.

Mm. Ideally find somewhere that isn't right next door to the tower's strongest entrenched position.

Kevak commented:
My thought is.. Maybe somewhere in the northeast ghetto region? There aren't much in the way of havens over there and it's rarely traveled by vampires. That or maybe near the skatepark or forest by the east tunnel? That's also fairly far from cammie territory, rarely traveled except by people who just spawned in, and is in the shadow of tzi territory, and as the tzi are aurtikis that the sabbat generally leave to their own devices/sometimes bug to tag along, that'd work quite well.

chazzyjazzy commented:
ok had some changes requested by xeon including :
removing antagonist classification for the roles as it was unneeded, this is done and i changed the add_antag_hud proc from being an antagonist only proc to a global proc with a different name, included a wrapper for good measure so that elsewhere where add_antag_hud is used the code doesnt explode and still uses the new global proc

datumizing ritae descriptions for the tome which is done

changing vaulderie stuff to align with another PR, not exactly done yet but it looks like making blood_donors a list of mob/living/carbon/human wasn't as bad as originally though rather than using real_names as that has problems with initiating the bloodbond (user_status_inlove)

antagonist name stuff is now deleted 

the blood refactor stuff im not exactly certain what must be changed as in general the code doesn't reference the blood reagant's properties a whole lot

and the masq contract showing the general area but not the GPS X and Y coordinates i'll do that tomorrow its probably an easy change

chazzyjazzy commented:
ok i think this is ready for another code review -

silver goblet's blood_donor list is storing mob/living/carbon/human and using that list to form bloodbonds
blood reagant is barely used so it shouldnt conflict with bloodpack pr
masq contract now shows general area for sabbat roles rather than GPS location
haven has been moved to a building in the ghetto northeast just below the coffee shop
antag name stuff deleted
antagonist classifications for priest and ductus are removed, add_antag_hud was reworked to be a global proc not antag only, with a wrapper proc so that anywhere in the code where add_antag_hud is used it refers to the new global proc

chazzyjazzy commented:
![dreammaker_5GX6kSbCXn](https://github.com/user-attachments/assets/8ef07740-1d09-4e98-8bdc-bab71a7760d6)

chazzyjazzy commented on tgstation.dme:
```diff
@@ -3498,11 +3501,15 @@
 #include "code\modules\wod13\datums\bouncer_roles\bouncer_mob_role.dm"
 #include "code\modules\wod13\datums\crafting_recipes\weed\weed_blunt.dm"
 #include "code\modules\wod13\datums\crafting_recipes\weed\weedpack.dm"
+#include "code\modules\wod13\datums\curses\__curse.dm"
```
dream maker fuckery

chazzyjazzy replied:
genuinely no clue how this is happening i only have one file for it hangon

chazzyjazzy replied:
oh wait it looks fixed? one is daimonion_curses and one is just _curse

chazzyjazzy commented:
potential problem : apparently burying a player fullkills them (and makes their body disappear), not sure how this works with my testing because in my video you can see the buried kindred emerges from the pit successfully, still something to test

chazzyjazzy commented on code/modules/wod13/ambient.dm:
```diff
@@ -324,6 +324,10 @@
 
+/area/vtm/sabbat_lair
+	name = "Interior"
+	zone_type = "elysium"
```
perhaps the spooky vtm:b music?

Nitro-Badger replied:
use some of the unused bloodlines music
https://youtu.be/TwLmV2EFjQ8?list=PLq33m2Vzy0f64hc2Vzc5AGVMVmfWFxWzS
https://youtu.be/NjVWGhKX4V0?list=PLq33m2Vzy0f64hc2Vzc5AGVMVmfWFxWzS
https://youtu.be/6Y3KfZ7Of-M?list=PLq33m2Vzy0f64hc2Vzc5AGVMVmfWFxWzS

buffyuwu commented on code/modules/admin/admin_verbs.dm:
```diff
@@ -990,7 +990,7 @@ GLOBAL_PROTECT(admin_verbs_hideable)
 	set category = "Admin.Game"
 
-	var/setting = input(usr, "Choose the bad guys party setting:", "Set Late Party") in list("caitiff", "sabbat", "hunter", "random")
+	var/setting = input(usr, "Choose the bad guys party setting:", "Set Late Party") in list("sabbat", "hunter", "random")
```
tgui inputs plz

buffyuwu commented on code/modules/food_and_drinks/drinks/drinks.dm:
```diff
@@ -214,6 +214,199 @@
+					M.apply_status_effect(STATUS_EFFECT_INLOVE, donor)
+					to_chat(M, span_warning("You feel a strange connection to <b>[donor]</b> forming as their blood mingles with yours!"))
+					to_chat(donor, span_notice("You sense that <b>[M]</b> has consumed your blood and is now bound to you."))
+					M.visible_message(span_notice("[M]'s eyes flash briefly as they become bound to [donor]."), "", span_notice("Your eyes flash as the blood bond forms."))
```
```suggestion
					M.visible_message(span_notice("[M]'s eyes flash briefly as they become bound to [donor]."), span_notice("Your eyes flash as the blood bond forms."))
```
i think positional arguments on visible_message are in order: message to everyone who can see it, message to the person its being called on. in this case, the first message plays to everyone nearby and the second plays to M

buffyuwu commented on code/modules/wod13/items/war_party.dm:
```diff
@@ -0,0 +1,55 @@
+		to_chat(H, span_cult("Only the Ductus or the Priest may call a War Party from the Totem!"))
+		return
+
+	var/choice = tgui_alert(H, "Do you wish to summon all Sabbat in the city to the Sabbat lair for the purposes of a War Party?", "War Party", list("Yes", "No"), 10 SECONDS)
```
```suggestion
	var/choice = tgui_alert(H, "Do you wish to send a message to all Sabbat in the city instructing them to return to the Sabbat lair?", "Return", list("Yes", "No"), 10 SECONDS)
```

XeonMations commented on code/modules/wod13/structures/doors/vampdoor_wood.dm:
```diff
@@ -43,3 +43,9 @@
 	lock_id = "cappadocian"
 	lockpick_difficulty = 8
+
+obj/structure/vampdoor/wood/sabbat
```
relatively pathed type defined here

XeonMations replied:
```suggestion
/obj/structure/vampdoor/wood/sabbat
```

chazzyjazzy commented:
@XeonMations conflict resolved

XeonMations commented:
No issues reported. Merging.
